### ISSUE TITLE
Latest LTS patch version not available on flathub

### ISSUE THREAD

**Issue by MattF-NSIDC** [flathub/org.qgis.qgis#329]
Currently the latest LTS release is 3.28.9, but 3.28.6 is the latest LTS available on flathub.

I don't know anything about contributing to the flatpak ecosystem yet, but I'm hoping to find time to learn soon and maybe submit a PR to address this.

Looks like [this](https://github.com/flathub/org.qgis.qgis/commit/d4c61ef89cecf03f9cd2ec862a1e9c67c428afd3) was the last change to the LTS version.

**Comment by steko** [comment]
The current LTS version is 3.28.13 but the app shows as "no longer receiving updates".

**Comment by MattF-NSIDC** [comment]
Thanks @aleixpol! Looking at #331 I think I could open a new PR to make 3.28.13 available. I have to find free time for it though.

When I first opened the issue I was being funded by a grant that would support work on this, but that's not the case anymore (until maybe March 2024).

**Comment by Neluji** [comment]
Hi, it actually looks like the LTS version [has to be moved to another branch](https://github.com/flathub/org.qgis.qgis/pull/396#issuecomment-1845444589) and EOL this branch for the build to work again (currently the version on flathub is 3.28.9).
As far as I understood the process, you (@aleixpol if you have the rights to do so) need to:

1. Move the code to the new branch,
2. EOL this one by adding [the appropriate key](https://docs.flathub.org/docs/for-app-authors/maintenance#end-of-life) in [`flathub.json`](https://github.com/flathub/org.qgis.qgis/blob/branch/lts/flathub.json),
3. and open an issue in [Flathub repo](https://github.com/flathub/flathub/issues) to ask to move the branch.

If all goes well it doesn't looks like such a hassle, and it will be nice to have the latest LTS version on Flathub ^^ (which is actually 3.28.14)

**Comment by sonicnkt** [comment]
Sorry to push this up again but it would be really great to have an up to date LTS flatpak again.

The Fedora COPR is not maintained any more either and this was a good solution to keep using the LTS Version for compatibility with other users.

**Comment by mfisher87** [comment]
I attempted to debug the flathub build and it seems the flathub build site is buggy or broken. The failed build for .13 is here:

https://buildbot.flathub.org/#/builders/6/builds/86119

When I click to view stdio or download it, it says there are 17 lines, but the fetched data is 0 bytes. @aleixpol can you suggest any steps for getting around this? If we can't view the build logs, I don't see a way to fix this package.

**Comment by mfisher87** [comment]
I can see logs in newer builds, e.g this unrelated project: https://buildbot.flathub.org/#/builders/6/builds/110954

Perhaps the issue is that Flathub archived the logs and isn't telling us that. Not sure how to re-run the build.

**Comment by mfisher87** [comment]
@Neluji Thanks for your post above! I didn't grok on first read but after more docs-diving I'm on the same page with you now.

As a person without maintainer access, how can I contribute to getting this process started? To the best of my knowledge, it's not possible to propose creation of a new branch through the PR interface. I'll start by creating a new issue on flathub/flathub.

**Comment by mfisher87** [comment]
@aleixpol looks like the only way to restore the LTS build is to follow this advice:

https://github.com/flathub/flathub/issues/5108#issuecomment-2028524586

> Applications aren't supposed to use branches other than `master` and `beta`, maybe the maintainer can make LTS available on master and move the current version from master to beta?

What do you think the best way forward is here? This would potentially be disruptive to users, but I think it's also important to provide LTS (they use the term LTR) as an option for installing QGIS, and dropping LTS support seems to be the only alternative.

**Comment by mfisher87** [comment]
Some more options were proposed in the above-linked thread.

**Comment by sonicnkt** [comment]
This is just a copy of my answer in the linked thread:

I think having two different releases on flathub (`org.qgis.qgis-ltr` and `org.qgis.qgis`) would be the best solution if flathub limits the channelss to standard and beta.

Having LTR on beta would make no sense and moving the stable to beta would cause even more confusion as you would essentially downgrade users which would result in projects files not opening correctly anymore.

The LTR version is a quite established name. Having the same icon should not be an issue if the name clearly indicates the version : "QGIS Desktop" vs" QGIS Desktop - LTR" .

This would also have the benefit to have both version easily installed side by side. Something that is not that unusual imo.
For work related projects i use the LTR since that is the version we deploy at the office for more stability. Sometimes i want to play arround with the latest features though.

**Comment by mfisher87** [comment]
That's very convincing, I think I agree with you.

I _love_ the idea of having both LTR and latest stable installed side-by-side, as that way I can better assess future impact to QGreenland!

**Comment by hellpe** [comment]
Is it because of this issue that the LTR flatpak is currently in version 3.28.9 instead of 3.34.15 [as advertised on the website](https://qgis.org/download/)?

**Comment by mfisher87** [comment]
That's my understanding. I haven't looked at this in a little while 😬

**Comment by PerAstraAdDeum** [comment]
> That's very convincing, I think I agree with you.
> 
> I _love_ the idea of having both LTR and latest stable installed side-by-side, as that way I can better assess future impact to QGreenland!
> 
> I'd be really interested to hear reasons not to do that. Already raised was the issue of a confusing application icon, and [@sonicnkt](https://github.com/sonicnkt) responded above ☝️

I agree. Having a LTS/LTR version on Flathub (ideally verified) would be the best way to handle this.

**Comment by PerAstraAdDeum** [comment]
Hello,

QGIS Flatpak has recently [been verified](https://github.com/flathub/org.qgis.qgis/issues/867). In my opinion, this is a major step toward greater visibility in the professional world.

I agree with @sonicnkt and @mfisher87 in that the most elegant solution would be to have a separate branch for up-to-date LTS (ideally verified as well).

As evident by issues like [this](https://github.com/flathub/org.qgis.qgis/issues/1044), the issue is still pending.
